cypress/e2e/registration.cy.js

```diff
@@ -1,11 +1,95 @@
 /// <reference types='cypress' />
+let user;
 
 describe('Student Registration page', () => {
   before(() => {
-
+    cy.visit('/automation-practice-form');
+    cy.task('generateUser').then((u) => { user = u; });
   });
 
   it('', () => {
+    cy.get('#firstName')
+      .type(user.firstName);
+
+    cy.get('#lastName')
+      .type(user.lastName);
+
+    cy.get('#userEmail')
+      .type(user.email);
+
+    cy.contains('.custom-control', user.gender)
+      .click();
+
+    cy.get('#userNumber')
+      .type(user.mobileNumber);
+
+    cy.get('#dateOfBirthInput')
+      .click();
+
+    cy.pickDate('month-select')
+      .select(user.birth.month);
+
+    cy.pickDate('year-select')
+      .select(`${user.birth.year}`);
+
+    cy.pickDate('day')
+      .contains(user.birth.day).click();
+
+    cy.get('.subjects-auto-complete__value-container')
+      .type('art{enter}' + 'his{enter}');
+
+    cy.contains('.custom-control-label', user.hobbies)
+      .click();
+
+    cy.get('#currentAddress')
+      .type(user.address);
+
+    cy.get('#state')
+      .type('{downarrow}{enter}');
+
+    cy.get('#city')
+      .type('{downarrow}{enter}');
+
+    cy.get('#submit')
+      .click();
+
+    cy.contains('tr', user.firstName)
+      .should('exist');
+
+    cy.get('table tbody tr')
```

Comment:
Please be more specific when you choose a selector. Now we should check that the concrete row in the table contains some data.
Fix it in further checks, please

Reply:
fixed